solr/core/src/java/org/apache/solr/core/SolrCores.java

```diff
@@ -114,11 +94,8 @@ protected void close() {
       synchronized (modifyLock) {
         // make a copy of the cores then clear the map so the core isn't handed out to a request
         // again
-        coreList.addAll(cores.values());
+        coreList.addAll(getCores());
```

Comment:
if this is already holding 'modifyLock' does it still need to call getCores which also attempts to grab the lock?

Reply:
There's no harm in synchronizing twice -- it's not worth trying to avoid it (e.g. perf).